src/main/java/de/tum/in/www1/artemis/service/TextExerciseFeedbackService.java

```diff
@@ -0,0 +1,171 @@
+        this.resultWebsocketService = resultWebsocketService;
+        this.participationService = participationService;
+        this.textSubmissionService = textSubmissionService;
+    }
+
+    private void checkRateLimitOrThrow(StudentParticipation participation) {
+
+        List<Result> athenaResults = participation.getResults().stream().filter(result -> result.getAssessmentType() == AssessmentType.AUTOMATIC_ATHENA).toList();
+
+        long countOfAthenaResultsInProcessOrSuccessful = athenaResults.stream().filter(result -> result.isSuccessful() == null || result.isSuccessful() == Boolean.TRUE).count();
+
+        long countOfSuccessfulRequests = athenaResults.stream().filter(result -> result.isSuccessful() == Boolean.TRUE).count();
+
+        // TODO For testing purposes, will be configurable in the future
+        if (countOfAthenaResultsInProcessOrSuccessful >= 10) {
+            throw new BadRequestAlertException("Cannot send additional AI feedback requests now. Try again later!", "participation", "preconditions not met");
+        }
+        if (countOfSuccessfulRequests >= 10) {
+            throw new BadRequestAlertException("Maximum number of AI feedback requests reached.", "participation", "preconditions not met");
+        }
+    }
+
+    /**
+     * Handles the request for generating feedback for a text exercise.
+     * Unlike programming exercises a tutor is not notified if Athena is not available.
+     *
+     * @param exerciseId    the id of the programming exercise.
+     * @param participation the student participation associated with the exercise.
+     * @param textExercise  the text exercise object.
+     * @return StudentParticipation updated text exercise for an AI assessment
+     */
+    public StudentParticipation handleNonGradedFeedbackRequest(Long exerciseId, StudentParticipation participation, TextExercise textExercise) {
+        if (this.athenaFeedbackSuggestionsService.isPresent()) {
+            this.checkRateLimitOrThrow(participation);
+            CompletableFuture.runAsync(() -> this.generateAutomaticNonGradedFeedback(participation, textExercise));
+        }
+        return participation;
+    }
+
+    /**
+     * Generates automatic non-graded feedback for a programming exercise submission.
+     * This method leverages the Athena service to generate feedback based on the latest submission.
+     *
+     * @param participation the student participation associated with the exercise.
+     * @param textExercise  the text exercise object.
+     */
+    public void generateAutomaticNonGradedFeedback(StudentParticipation participation, TextExercise textExercise) {
+        log.debug("Using athena to generate (text exercise) feedback request: {}", textExercise.getId());
+
+        // athena takes over the control here
+        var submissionOptional = participationService.findTextExerciseParticipationWithLatestSubmissionAndResult(participation.getId()).findLatestSubmission();
```

Comment:
Wouldn't it be enough to retrieve the latest submission from the database?
Please also use an `ElseThrow`-variant in the repository instead of checking here if a submission is present